### ISSUE TITLE
Insert add-on ID into manifest if it's missing

### ISSUE THREAD

**Issue by jvillalobos** [mozilla/addons-server#11559]
One of the issues that came up during Armagadd-on is that it's problematic for add-ons to not have an ID set in their manifest. A broken signature means there's no way for Firefox to figure out their ID.

AMO should add the [ID in the manifest](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/browser_specific_settings) if the uploaded add-on doesn't have it, before signing.

The reason we don't want to require developers to add the ID themselves is that we don't want to make it harder for them to maintain a Firefox and Chrome version of their add-on. In some cases it's possible to submit the same code for both browsers, so we should try to keep things that way if possible.

/cc @muffinresearch @ddurst

**Comment by eviljeff** [comment]
QA: this needs the waffle switch `add-guid-to-manifest` (adding and) enabling to test - and shouldn't rewrite if the waffle is disabled.

**Comment by ioanarusiczki** [comment]
@eviljeff 

I've submitted on AMO dev 2 addons with no "id" in their manifest.json file , also `[add-guid-to-manifest](https://addons-internal.dev.mozaws.net/en-US/admin/models/waffle/switch/114/change/)` is active now on -dev. 

I don't see an "id" in their manifest.json, these are the 2 samples:
https://code.addons-dev.allizom.org/en-US/browse/621275/versions/1693690/?path=manifest.json
https://code.addons-dev.allizom.org/en-US/browse/621273/versions/1693688/?path=manifest.json

**Comment by eviljeff** [comment]
> @eviljeff
> 
> I've submitted on AMO dev 2 addons with no "id" in their manifest.json file , also [add-guid-to-manifest](https://addons-internal.dev.mozaws.net/en-US/admin/models/waffle/switch/) is active now on -dev.

The id is added just before signing so if it's not signed it won't be there yet

**Comment by ioanarusiczki** [comment]
@eviljeff 
because of 

> AMO should add the [ID in the manifest](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/browser_specific_settings) if the uploaded add-on doesn't have it, before signing. 

I understood it should be available before signing. 

But I got it now, I manually approved the first version than downloaded the .xpi and I could see the `id`. I'll check around.  
I assume this applies for addons submitted using the API too and for all add-on types.

**Comment by eviljeff** [comment]
yes, it should be all add-ons - we sign them all now (except the pre-signed submission that were already signed with the Mozilla privileged certificate)

**Comment by ioanarusiczki** [comment]
@eviljeff The switch seems to be working as expected for:
- extensions 
- static themes
- language packs
- Dicts cannot be submitted without an id -> so they're an exception to the general rule. 

What I've noticed is that the layout of the code is changed after the "id" is inserted in manifest.json and that could not be an expected outcome.  This is reproducible for all addons I've tested with.

An example below, before submission: 

![code layout](https://user-images.githubusercontent.com/33448286/156165710-71d3f7d8-e6ab-439f-b0dc-295ffb9e8671.png)

After "id" is part of the manifest.json file , code has one line: 

![new layout](https://user-images.githubusercontent.com/33448286/156165846-248bd4c5-be02-4b53-8e56-8fdd04608de0.png)

**Comment by eviljeff** [comment]
> What I've noticed is that the layout of the code is changed after the "id" is inserted in manifest.json and that could not be an expected outcome. This is reproducible for all addons I've tested with.
?
> After "id" is part of the manifest.json file , code has one line:

can you file a follow-up issue?

**Comment by willdurand** [comment]
> `json.dumps()` in [#18875 (files)](https://github.com/mozilla/addons-server/pull/18875/files#diff-0e760bc678a5fb093117b518b9badc6e3259826bfb89517ca530d6303c5d0e88R91) is missing `indent=2` as a parameter.

It'd be good to introduce a shared function since it is also used in https://github.com/mozilla/addons-server/blob/3c117d2dea0559c6de5b32a949089923b10e64ff/src/olympia/files/tasks.py#L60

**Comment by ioanarusiczki** [comment]
@eviljeff Done https://github.com/mozilla/addons-server/issues/18920
